sdk/schemaregistry/Azure.Data.SchemaRegistry/src/SchemaFormat.cs

```diff
@@ -63,37 +71,34 @@ internal ContentType ToContentType()
             switch (_value)
             {
                 case AvroValue:
-                    return ContentType.Avro;
+                    return new ContentType(AvroContentType);
                 case JsonValue:
-                    return ContentType.Json;
+                    return new ContentType(JsonContentType);
                 //case ProtobufValue:
-                //    return ContentType.Protobuf;
+                //    return new ContentType(ProtobufContentType);
                 default:
-                    return ContentType.Custom;
+                    return new ContentType(CustomContentType);
             }
         }
 
         internal static SchemaFormat FromContentType(string contentTypeValue)
         {
             var contentTypeParameterValue = contentTypeValue.Split('=');
-            var contentSubType = contentTypeValue.Split('/');
             if (contentTypeParameterValue.Length > 1)
             {
                 switch (contentTypeParameterValue[1])
                 {
-                    case AvroContentType:
-                        return SchemaFormat.Avro;
-                    case JsonContentType:
-                        return SchemaFormat.Json;
+                    case AvroContentTypeValue:
+                        return Avro;
+                    case JsonContentTypeValue:
+                        return Json;
+                    case CustomContentTypeValue:
+                        return Custom;
                     default:
                         break;
                 }
             }
-            return contentSubType[1] switch
-            {
-                //ProtobufContentType => SchemaFormat.Protobuf,
-                _ => SchemaFormat.Custom,
-            };
+            return new SchemaFormat(contentTypeValue);
```

Comment:
this looks wrong to me.   Don't we want to return `contentTypeParameterValue[1]` when the split was successful?

Reply:
The new flow is that it splits on the '=' and if what's after the '=' matches "Avro", "Json", or "utf-8" the corresponding format is returned (Avro, JSON, or Custom, respectively). If it's not one of those three then there isn't enough information to know how to interpret or split the value, it could be anything, so we use the whole content type.

Reply:
I think this is what we agreed to do in the chat, just return the content type string if we don't know what it is

Reply:
Ah, I see where I went wrong.   I confused `AvroContentType` with `AvroContentTypeValue`.   All is well here.